### ISSUE TITLE
Elasticsearch integration 

### ISSUE THREAD

**Issue by Malwar3Ninja** [MISP/MISP#6907]
Hi Team, 

I wanted to check if there was a way to easily get my IOC's from MISP to elasticsearch ?

Thanks

**Comment by cudeso** [comment]
I used the Filebeat module to push IOCs to Elastic. See: https://www.vanimpe.eu/2021/01/13/use-elastic-to-represent-misp-threat-data/

Also have a look here for enrichments: https://www.securitydistractions.com/2019/05/17/enriching-elasticsearch-with-threat-data-part-1-misp/

**Comment by Malwar3Ninja** [comment]
> I used the Filebeat module to push IOCs to Elastic. See: https://www.vanimpe.eu/2021/01/13/use-elastic-to-represent-misp-threat-data/
> 
> Also have a look here for enrichments: https://www.securitydistractions.com/2019/05/17/enriching-elasticsearch-with-threat-data-part-1-misp/

Yup I used the filebeat module but the there is an issue for duplicates. I feel this is a temporary fix, 

I also synced my mysql db with elasticsearch using logstash jdbc plugin and it works fine if attributes table is just picked. Still trying to make a workaround about nested queries to get all feed data. Any help would be great.

**Comment by baonq-me** [comment]
> Hi Team,
> 
> I wanted to check if there was a way to easily get my IOC's from MISP to elasticsearch ?
> 
> Thanks

By using `Debezium` as a source connector, `Kafka connect`, and `kafka-connect-elasticsearch` as a sink, you can stream the table `misp.attributes` from MariaBD into `ElasticSearch`.

Ref:
https://github.com/confluentinc/kafka-connect-elasticsearch
https://debezium.io/documentation/reference/stable/tutorial.html
https://sematext.com/blog/kafka-connect-elasticsearch-how-to/
